translator/src/main/java/com/google/devtools/j2objc/gen/GenerationUnit.java

```diff
@@ -206,16 +205,19 @@ public boolean isFullyParsed() {
   }
 
   /**
-   * Gets the output path if there isn't one already.
-   * For example, foo/bar/Mumble.java translates to $(OUTPUT_DIR)/foo/bar/Mumble.
-   * If --no-package-directories is specified, though, the output file is $(OUTPUT_DIR)/Mumble.
+   * Gets the output path if there isn't one already. For example, foo/bar/Mumble.java translates to
+   * $(OUTPUT_DIR)/foo/bar/Mumble. If --no-package-directories is specified, though, the output file
+   * is $(OUTPUT_DIR)/Mumble.
    */
   private static String getDefaultOutputPath(CompilationUnit unit) {
     String path = unit.getMainTypeName();
     PackageDeclaration pkg = unit.getPackage();
     if (Options.usePackageDirectories() && !pkg.isDefaultPackage()) {
       path = pkg.getName().getFullyQualifiedName().replace('.', File.separatorChar)
           + File.separatorChar + path;
+    } else if (Options.generatePackagePrefixedFilenames()) {
+      // Prepend the package in camelcase to the class
+      path = NameTable.camelCaseQualifiedName(pkg.getName().getFullyQualifiedName()) + path;
```

Comment:
Use nameTable.getPrefix(pkg). This will incorporate any package prefix renaming into your file name.